pkg/hooks/webhook.go

```diff
@@ -83,7 +119,18 @@ func (w *WebhookExecutor) Call(request interface{}, response interface{}) error
 		rawRequest := json.RawMessage(reqBody)
 		logging.Logger.Info("Webhook request", "type", w.hookType, "url", w.url, "body", rawRequest)
 	}
-	resp, err := w.client.Post(w.url, "application/json", bytes.NewReader(reqBody))
+
+	var cacheEntry = w.etagCache.Get(request.GetCacheKey())
```

Comment:
suggest: This can be move to later parts of the code. E.g. we do not need this entry that involves lock if response failed.

Reply:
We need this entry before we run query, since if it is there we need to include If-none-match header with cached etag